.github/workflows/pythonpublish.yml

```diff
@@ -0,0 +1,26 @@
+name: Upload Python Package
+
+on:
+  release:
+    types: [created]
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v1
```

Comment:
Nit: Maybe this step also deserves its name to be consistent with the rest of the steps?

Reply:
Which branch gets checked out? The default branch?

Reply:
A name can be added to this step for added clarity :+1:

Reply:
Fair question about which version of the code gets used, took me a bit to find more info on that when I first was getting into it. Basically when a github action gets triggered it will use the git_sha given by that trigger. In the case of the release trigger, it will use the git_sha in the release which should correspond to the version we want to release.

If you want to read more into here, I found the following articles helpful:
General info on workflow triggers - https://help.github.com/en/actions/automating-your-workflow-with-github-actions/events-that-trigger-workflows#about-workflow-events 
Release trigger - https://help.github.com/en/actions/automating-your-workflow-with-github-actions/events-that-trigger-workflows#release-event-release